docs/development/releasing.rst

```diff
@@ -1,233 +1,386 @@
-      maintainers about their relevant pull requests, so that the maintainers
-      have the option to re-milestone their work.
+      `What's New in Astropy <current_version>?
+      <https://docs.astropy.org/en/v<current_version>/whatsnew/<current_version>.html>`__
 
-#. If there are any issues in the GitHub issue tracker that are labeled
-   ``affects-dev`` but are issues that apply to this release, update them to
-   ``affects-release``.  Similarly, if any issues remain open for this release,
-   re-assign them to the next relevant milestone.
+   This is because we want to make sure that links in the previous "What's new?" pages continue
+   to work and reference the original link they referenced at the time of writing.
 
-#. (astropy specific) Ensure the built-in IERS earth rotation parameter and
-   leap second tables are up to date by changing directory to
-   ``astropy/utils/iers/data`` and executing ``update_builtin_iers.sh``.
-   Check the result with ``git diff`` (do not be surprised to find many lines
-   in the ``eopc04_IAU2000.62-now`` file change; those data are reanalyzed
-   periodically) and committing. Since in some cases updating the IERS tables
-   may result in test failures, this update should be done via a pull request
-   to the ``main`` branch, and then backported to the release branch.
+#. Commit these changes::
 
-#. (Only for major versions) Make sure to update the "What's new"
-   section with the stats on the number of issues, PRs, and contributors.
-   Since the What's New for the major release is now only present in the release
-   branch, you should switch to it to, e.g.::
+      $ git add docs/whatsnew/<current_version>.rst
+      $ git add docs/whatsnew/<next_version>.rst
+      $ git add docs/whatsnew/index.rst
+      $ git add docs/index.rst
+      $ git commit -m "Added <next_version> what's new page and redirect <current_version> what's new page"
 
-      $ git checkout v5.0.x
+#. Tag this commit using the next major version followed by ``.dev``. For example,
+   if you have just branched ``v5.0.x``, create the ``v5.1.dev`` tag::
 
-   To find the statistics and contributors, use the `generate_releaserst.xsh`_
-   script. This requires `xonsh <https://xon.sh/>`_ and `docopt
-   <http://docopt.org/>`_ which you can install with::
+      $ git tag -s "v<next_version>.dev" -m "Back to development: v<next_version>"
 
-      pip install xonsh docopt
+#. Push all of these changes up to github::
 
-   You should then run the script in the root of the astropy repository as follows::
+      $ git push upstream v<version>.x:v<version>.x
+      $ git push upstream main:main
 
-      xonsh generate_releaserst.xsh 4.3 v5.0.dev \
-                                    --project-name=astropy \
-                                    --pretty-project-name=astropy \
-                                    --pat=<a GitHub personal access token>
+#. On the GitHub issue tracker, add a new milestone for the next major version
```

Comment:
I think the milestone creation should be moved to the top to be an early step at the time of announcing the feature freeze. So maintainers can remilestone PRs that won't make the release.

Reply:
Done